packages/browser-sdk/src/lib/helpers/hooks/useWindowSize.ts

```diff
@@ -0,0 +1,49 @@
+import * as React from "react";
+
+import { debounce } from "@whereby.com/core";
+import { captureSafeAreaInsets } from "../safeAreaInsets";
+
+const captureDocumentSize = () => {
+    return {
+        width: document.documentElement.clientWidth,
+        height: document.documentElement.clientHeight,
+    };
+};
+
+type Dimensions = {
+    windowSize: { width: number; height: number };
+    safeAreaInsets: { top: number; right: number; bottom: number; left: number };
+};
+
+const dimensionsReducer = (state: Dimensions, action: { type: "update"; newDimensions: Dimensions }) => {
+    switch (action.type) {
+        case "update":
+            return {
+                ...state,
+                ...action.newDimensions,
+            };
+        default:
+            return state;
+    }
+};
+
+const captureDimensions = () => ({
+    windowSize: captureDocumentSize(),
+    safeAreaInsets: captureSafeAreaInsets(),
+});
+
+export const useWindowSize = (isTouchDevice: boolean) => {
+    const [state, dispatch] = React.useReducer(dimensionsReducer, {}, captureDimensions);
```

Comment:
Can we not just use something very simple like this here and remove redux?

```js
export const useWindowSize = (isTouchDevice: boolean) => {
    const [dimensions, setDimensions] = React.useState(captureDimensions());

    React.useEffect(() => {
        const handler = () => {
            const newDimensions = captureDimensions();
            setDimensions(newDimensions)
        };
        const onResize = isTouchDevice ? handler : debounce(handler, { delay: 60 });
        window.addEventListener("resize", onResize);
        return () => window.removeEventListener("resize", onResize);
    }, [isTouchDevice]);

    return dimensions;
}
```

Reply:
Yeah, good call. Changed now